torch/distributed/checkpoint/filesystem.py

```diff
@@ -358,6 +363,20 @@ def rename(
     def mkdir(self, path: Union[str, os.PathLike]) -> None:
         cast(Path, path).mkdir(parents=True, exist_ok=True)
 
+    @classmethod
+    def validate_checkpoint_id(cls, checkpoint_id: Union[str, os.PathLike]) -> bool:
+        if isinstance(checkpoint_id, Path):
+            return True
+
+        if "://" in str(checkpoint_id):
+            return False
+
+        for p in Path(checkpoint_id).parents:
+            if p.exists() and os.access(str(p), os.W_OK):
+                return True
```

Comment:
Why is it necessary to require the path to be writable?
If the path is read-only, users should be allowed to use DCP to read checkpoints.